### PR TITLE
fix(highlight): add create param in nvim_get_hl api function

### LINKED ISSUE #24583: `nvim_get_hl()` creates undefined highlight groups
### Problem

`nvim_get_hl(<ns>, { name = 'Foo' })` creates a cleared highlight group `hl-Foo` if `hl-Foo` does not exist, which is an undocumented side effect.

This affects regions highlighted by non-existing highlight groups in a winbar: consider a winbar set to `%#Foo#abcd%*`, if `hl-Foo` does not exist, highlight of `abcd` will fall back to `hl-WinBar` or `hl-WinBarNC`, however if `hl-Foo` is a cleared highlight group, the color of `abcd` will fall back to `hl-Normal`.

### Steps to reproduce

1. nvim --clean
2. `:echo hlexists('Foo')` --> 0
3. `:call nvim_get_hl(0, { 'name' = 'Foo' })`
4. `:echo hlexists('Foo')` --> 1

### Expected behavior

1. nvim --clean
2. `:echo hlexists('Foo')` --> 0
3. `:call nvim_get_hl(0, { 'name' = 'Foo' })`
4. `:echo hlexists('Foo')` --> 0

### Neovim version (nvim -v)

NVIM v0.10.0-dev-751+gd086bc1e85

### Vim (not Nvim) behaves the same?

no, vim does not have `nvim_get_hl()`

### Operating system/version

Linux 6.1.39-2-lts

### Terminal name/version

wezterm 20230709-182848-f901dfbf

### $TERM environment variable

tmux-256color

### Installation

AUR

### PR DESCRIPTION
**Pull request by glepnir**
Problem: `syn_check_group` will auto create a new highlight group when it not exist when using `nvim_get_hl` , Sometimes it's annoying.

Solution: add a flag `create` in `nvim_get_hl` default is true, set to false disable auto create when highlight group not exist.

Fix #24583

### CODE REVIEW

**Comment by bfredl** [comment]
This is somewhat intentional, as some use cases _need_ to refer to highlight groups before the user adds highlights to them. We could add an option to disable it tho.

File: src/nvim/highlight_group.c [hunk 1]
```diff
@@ -1568,7 +1568,16 @@ Dictionary ns_get_hl_defs(NS ns_id, Dict(get_highlight) *opts, Arena *arena, Err
   Boolean link = GET_BOOL_OR_TRUE(opts, get_highlight, link);
   int id = -1;
   if (HAS_KEY(opts, get_highlight, name)) {
-    id = syn_check_group(opts->name.data, opts->name.size);
+    Boolean create = false;
```
Comment (bfredl) [inline src/nvim/highlight_group.c:1571]: can use `GET_BOOL_OR_TRUE`

File: src/nvim/highlight_group.c [hunk 2]
```diff
@@ -1568,7 +1568,16 @@ Dictionary ns_get_hl_defs(NS ns_id, Dict(get_highlight) *opts, Arena *arena, Err
...
+    if (HAS_KEY(opts, get_highlight, create)) {
+      create = opts->create;
+    }
+    id = create ? syn_check_group(opts->name.data, opts->name.size)
+                : syn_name2id_len(opts->name.data, opts->name.size);
+    if (id == 0 && strchr(opts->name.data, ' ') == NULL) {
```
Comment (bfredl) [inline src/nvim/highlight_group.c:1577]: This is a bit specific, `' '` is not the only invalid char. could check `id =≠ 0 && !create` instead.

**Comment by glepnir** [comment]
done

File: src/nvim/api/vim.c [hunk 3]
```diff
@@ -96,6 +96,7 @@ Integer nvim_get_hl_id_by_name(String name)
 ///                 - name: (string) Get a highlight definition by name.
 ///                 - id: (integer) Get a highlight definition by id.
 ///                 - link: (boolean, default true) Show linked group name instead of effective definition |:hi-link|.
+///                 - create: (boolean, default true) when highlight group not exist create it
```
Comment (zeertzjq) [inline src/nvim/api/vim.c:99]:
```suggestion
///                 - create: (boolean, default true) When highlight group doesn't exist create it.
```

**Comment by zeertzjq** [comment]
Needs rebase and regeneration of docs.

File: src/nvim/highlight_group.c [hunk 4]
```diff
@@ -1569,7 +1569,16 @@ Dictionary ns_get_hl_defs(NS ns_id, Dict(get_highlight) *opts, Arena *arena, Err
   Boolean link = GET_BOOL_OR_TRUE(opts, get_highlight, link);
   int id = -1;
   if (HAS_KEY(opts, get_highlight, name)) {
-    id = syn_check_group(opts->name.data, opts->name.size);
+    Boolean create = GET_BOOL_OR_TRUE(opts, get_highlight, create);
+    if (HAS_KEY(opts, get_highlight, create)) {
+      create = opts->create;
+    }
```
Comment (bfredl) [inline src/nvim/highlight_group.c:1575]: this is redundant and can be removed (`GET_BOOL_OR_TRUE` does everything)

Comment (glepnir) [reply src/nvim/highlight_group.c]: done.

**Pull request merged**